File: python/mirascope/llm/providers/anthropic/model_id.py
"""Anthropic registered LLM models."""

from typing import TypeAlias, get_args

from .model_info import AnthropicKnownModels

AnthropicModelId: TypeAlias = AnthropicKnownModels | str
"""The Anthropic model ids registered with Mirascope."""

ANTHROPIC_KNOWN_MODELS: set[str] = set(get_args(AnthropicKnownModels))


def model_name(model_id: AnthropicModelId) -> str:
    """Extract the anthropic model name from the ModelId

    Args:
        model_id: Full model ID (e.g. "anthropic/claude-sonnet-4-5" or
            "anthropic-beta/claude-sonnet-4-5")

    Returns:
        Provider-specific model ID (e.g. "claude-sonnet-4-5")
    """
    return model_id.removeprefix("anthropic-beta/").removeprefix("anthropic/")

File: typescript/src/llm/providers/anthropic/model-id.ts
/**
 * Anthropic registered LLM models.
 */

import type { AnthropicKnownModels } from '@/llm/providers/anthropic/model-info';

/**
 * The Anthropic model IDs registered with Mirascope.
 */
export type AnthropicModelId = AnthropicKnownModels | (string & {});

/**
 * Extract the Anthropic model name from the ModelId.
 *
 * @param modelId - Full model ID (e.g. "anthropic/claude-sonnet-4-5" or "anthropic-beta/claude-sonnet-4-5")
 * @returns Provider-specific model ID (e.g. "claude-sonnet-4-5")
 */
export function modelName(modelId: AnthropicModelId): string {
  return modelId.replace(/^anthropic-beta\//, '').replace(/^anthropic\//, '');
}
